src/Database/Schema/SchemaDialect.php

```diff
@@ -182,13 +182,31 @@ public function dropTableSql(TableSchema $schema): array
 
     /**
      * Generate the SQL to list the tables.
-     *
+     * @deprecated
      * @param array<string, mixed> $config The connection configuration to use for
      *    getting tables from.
      * @return array An array of (sql, params) to execute.
      */
     abstract public function listTablesSql(array $config): array;
 
+    /**
+     * Generate the SQL to list the tables, excluding all views.
+     *
+     * @param array<string, mixed> $config The connection configuration to use for
+     *    getting tables from.
+     * @return array An array of (sql, params) to execute.
+     */
+    abstract public function listTablesWithoutViewsSql(array $config): array;
```

Comment:
We will have to make these annotations as well to not break dialects in userland code. Currently they will fatal error about the missing methods.